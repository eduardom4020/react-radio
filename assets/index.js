import RussiaBG from './russia.jpg';
import DribbbledBG from './dribbbled.jpg';
import DogeBG from './doge.jpg';
import BalladsBG from './ballads.jpg';
import MaximumBG from './maximum.jpg';

import Switch from './switch.png';
import Back from './back-arrow.png';

export const radioPutinLogo = RussiaBG;
export const radioDribbbledLogo = DribbbledBG;
export const radioDogeLogo = DogeBG;
export const radioBalladsLogo = BalladsBG;
export const radioMaximumLogo = MaximumBG;

export const iconSwitch = Switch;
export const iconBack = Back;