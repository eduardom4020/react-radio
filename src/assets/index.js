import RussiaBG from './russia.jpg';
import DribbbledBG from './dribbbled.jpg';
import DogeBG from './doge.jpg';
import BalladsBG from './ballads.jpg';
import MaximumBG from './maximum.jpg';

import Switch from './switch.png';
import Back from './back-arrow.png';
import Plus from './plus.png';
import Minus from './minus.png';

export const radioPutinLogo = RussiaBG;
export const radioDribbbledLogo = DribbbledBG;
export const radioDogeLogo = DogeBG;
export const radioBalladsLogo = BalladsBG;
export const radioMaximumLogo = MaximumBG;

export const iconSwitch = Switch;
export const iconBack = Back;
export const iconPlus = Plus;
export const iconMinus = Minus;