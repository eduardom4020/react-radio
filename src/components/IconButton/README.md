# IconButton
>This component is a button made by any image. It always keeps the aspect ration of that image, by creating only squared buttons.

### Props
These are the props accepted by the component, pay attention to the description, inputs and behaviour:

---

##### src *(image | string| object)*
*The image 'icon' showed in button. It can be an loaded image (using require or ES6 import), an relative path for image and an object containing key 'url' that have receives a string as value, and loads the image from internet to show in the application*. 

**example**: 
```
import Image from './path-to-image';
<IconButton src={Image} /> 
```
---

##### size *(string | number)*
*Sets the size of the button, by setting equal width and height in CSS. By default, it value is 24px.*

**ATTENTION**: you can make rectangular buttons by passing *style* props with width or height. In this case, the size value will be overridden. 

**example**:
```
<IconButton size={50} /> 
```

---