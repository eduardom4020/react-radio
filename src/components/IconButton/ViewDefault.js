import React from 'react';
import styled from 'styled-components';
import { Image, TouchableOpacity } from 'react-native';
import { DEFAULT_ICON_BT_SIZE } from '~/src/constants/values';

const StyledImage = styled(Image)`
    width: ${props => props.size || DEFAULT_ICON_BT_SIZE}px;
    height: ${props => props.size || DEFAULT_ICON_BT_SIZE}px;
`;

const IconButtonViewDefault = props => {
    const { src, ...extraProps } = props;

    return (
        <TouchableOpacity>
            <StyledImage
                source={src}
                {...extraProps}
            />
        </TouchableOpacity>
    );
};

export default IconButtonViewDefault;