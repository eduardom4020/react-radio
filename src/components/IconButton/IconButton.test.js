import React from 'react';
import { iconSwitch } from '~/assets';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import IconButton from '~/src/components/IconButton';

describe('IconButton', () => {
    test('Render?', () => {
        const component = shallow(
            <IconButton />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with custom size?', () => {
        const component = shallow(
            <IconButton size={64} />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with image from assets?', () => {
        const component = shallow(
            <IconButton src={iconSwitch} />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with invalid image source type?', () => {
        const componentObject = shallow(
            <IconButton src={{}} />
        );

        expect(toJson(
            componentObject, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
        
        const componentBool = shallow(
            <IconButton src={true} />
        );

        expect(toJson(
            componentBool, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();

        const componentNumber = shallow(
            <IconButton src={1} />
        );

        expect(toJson(
            componentNumber, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();

        const componentString = shallow(
            <IconButton src='image' />
        );

        expect(toJson(
            componentString, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
});