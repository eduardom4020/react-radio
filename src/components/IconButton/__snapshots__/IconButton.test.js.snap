// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Render with custom size? 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <Styled(Image)
    size={64}
  />
</TouchableOpacity>
`;

exports[`IconButton Render with image from assets? 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <Styled(Image)
    source={
      Object {
        "testUri": "../../../assets/switch.png",
      }
    }
  />
</TouchableOpacity>
`;

exports[`IconButton Render with invalid image source type? 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <Styled(Image)
    source={Object {}}
  />
</TouchableOpacity>
`;

exports[`IconButton Render with invalid image source type? 2`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <Styled(Image)
    source={true}
  />
</TouchableOpacity>
`;

exports[`IconButton Render with invalid image source type? 3`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <Styled(Image)
    source={1}
  />
</TouchableOpacity>
`;

exports[`IconButton Render with invalid image source type? 4`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <Styled(Image)
    source="image"
  />
</TouchableOpacity>
`;

exports[`IconButton Render? 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <Styled(Image) />
</TouchableOpacity>
`;
